pkg/podmounter/mppod/creator.go

```diff
@@ -251,16 +255,44 @@ func (c *Creator) createCacheVolumeSourceForEmptyDir(volumeAttributes map[string
 	return corev1.VolumeSource{EmptyDir: emptyDir}, nil
 }
 
-// createCacheVolumeSourceForEmptyDir creates an `persistentVolumeClaim` volume source to use as local-cache.
-func (c *Creator) createCacheVolumeSourceForPVC(volumeAttributes map[string]string) (corev1.VolumeSource, error) {
-	claimName := volumeAttributes[volumecontext.CachePersistentVolumeClaimName]
-	if claimName == "" {
-		return corev1.VolumeSource{}, fmt.Errorf("%q must be provided with %q cache type", volumecontext.CachePersistentVolumeClaimName, cacheTypePersistentVolumeClaim)
+// createCacheVolumeSourceForEphemeral creates an `ephemeral` volume source to use as local-cache.
+func (c *Creator) createCacheVolumeSourceForEphemeral(volumeAttributes map[string]string) (corev1.VolumeSource, error) {
+	storageClassName := volumeAttributes[volumecontext.CacheEphemeralStorageClassName]
+	if storageClassName == "" {
+		return corev1.VolumeSource{}, fmt.Errorf("%q must be provided with %q cache type", volumecontext.CacheEphemeralStorageClassName, cacheTypeEphemeral)
 	}
 
-	return corev1.VolumeSource{PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-		ClaimName: claimName,
-	}}, nil
+	storageResourceRequestStr := volumeAttributes[volumecontext.CacheEphemeralStorageResourceRequest]
+	if storageResourceRequestStr == "" {
+		return corev1.VolumeSource{}, fmt.Errorf("%q must be provided with %q cache type", volumecontext.CacheEphemeralStorageResourceRequest, cacheTypeEphemeral)
+	}
+
+	storageRequestSize, err := resource.ParseQuantity(storageResourceRequestStr)
+	if err != nil {
+		return corev1.VolumeSource{}, failedToParseQuantityError(err, volumecontext.CacheEphemeralStorageResourceRequest, storageResourceRequestStr)
+	}
+
+	return corev1.VolumeSource{
+		Ephemeral: &corev1.EphemeralVolumeSource{
+			VolumeClaimTemplate: &corev1.PersistentVolumeClaimTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"type": "mountpoint-csi-driver-local-cache",
```

Comment:
Is this label used for anything? We usually set some prefix like this for our labels/annotations `s3.csi.aws.com/`?

Reply:
Updated with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/496/commits/b5216c7b894ba5b2c52f65a2a9e4e7d519eeb501